### DISCUSSION TITLE
about filter specified attachment's name

### DISCUSSION THREAD

**Discussion by Bailievers** [ecederstrand/exchangelib#1251] [General]
My requirement is to filter emails based on attachment 's name,
On the "account. inbox. filter" function.
I tried various "item: property" options but couldn't find a solution to meet my requirements.
The only use of 'filter (attachments=)' caught my attention
But it requires an instance of the Attachment class to be passed in, but I only need to search for attachments. Why create attachments?
Hope for help

**Comment by ecederstrand** [comment]
exchangelib supports [AQS](https://learn.microsoft.com/da-dk/windows/win32/lwef/-search-2x-wds-aqsreference?redirectedfrom=MSDN) which may solve your problem, but otherwise I don't think it's possible in EWS.

**Comment by Bailievers** [comment]
> exchangelib supports [AQS](https://learn.microsoft.com/da-dk/windows/win32/lwef/-search-2x-wds-aqsreference?redirectedfrom=MSDN) which may solve your problem, but otherwise I don't think it's possible in EWS.

Thank you for your reply
I just tried AQS and came back
Aqs' 'attachment:' cannot match the attachment file name, which surprised me
It can search for recognizable attachment content and also search for email body
Damn it, don't search for names
Helplessly, I can only download the email and then parse and filter it to meet my needs

**Comment by ecederstrand** [comment]
You could be lucky that the attachment name is also available in the `mime_content` field. It's worth a try, at least.
